src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -21,17 +18,12 @@ internal partial class DiagnosticAnalyzerService : IIncrementalAnalyzerProvider
     {
         public IIncrementalAnalyzer CreateIncrementalAnalyzer(Workspace workspace)
         {
-            if (GlobalOptions.IsLspPullDiagnostics())
-            {
-                // We rely on LSP to query us for diagnostics when things have changed and poll us for changes that might
-                // have happened to the project or closed files outside of VS.
-                // However, we still need to create the analyzer so that the map contains the analyzer to run when pull diagnostics asks.
-                _ = _map.GetValue(workspace, _createIncrementalAnalyzer);
-
-                return NoOpIncrementalAnalyzer.Instance;
-            }
+            var analyzer = _map.GetValue(workspace, _createIncrementalAnalyzer);
 
-            return _map.GetValue(workspace, _createIncrementalAnalyzer);
+            // We rely on LSP to query us for diagnostics when things have changed and poll us for changes that might
+            // have happened to the project or closed files outside of VS. However, we still need to create the analyzer
+            // so that the map contains the analyzer to run when pull diagnostics asks.
+            return GlobalOptions.IsLspPullDiagnostics() ? NoOpIncrementalAnalyzer.Instance : analyzer;
```

Comment:
same logic a before.  i just rejiggered.  in both paths below we were adding hte item to the map.  i pulled that above, and then just changed the logic if the NoOp analyzer is returned or the actual instance.